table/block_based/block_based_table_reader.cc

```diff
@@ -2911,25 +3041,43 @@ Status BlockBasedTable::Get(const ReadOptions& read_options, const Slice& key,
           // the end of the block, i.e. cannot be in the following blocks
           // either. In this case, the seek_key cannot be found, so we break
           // from the top level for-loop.
-          break;
-        }
+          done = true;
+        } else {
+          // Call the *saver function on each entry/block until it returns false
+          for (; biter.Valid(); biter.Next()) {
+            ParsedInternalKey parsed_key;
+            if (!ParseInternalKey(biter.key(), &parsed_key)) {
+              s = Status::Corruption(Slice());
+            }
 
-        // Call the *saver function on each entry/block until it returns false
-        for (; biter.Valid(); biter.Next()) {
-          ParsedInternalKey parsed_key;
-          if (!ParseInternalKey(biter.key(), &parsed_key)) {
-            s = Status::Corruption(Slice());
+            if (!get_context->SaveValue(
+                    parsed_key, biter.value(), &matched,
+                    biter.IsValuePinned() ? &biter : nullptr)) {
+              is_referenced_key_exist = true;
+              done = true;
+              break;
+            }
           }
-
-          if (!get_context->SaveValue(
-                  parsed_key, biter.value(), &matched,
-                  biter.IsValuePinned() ? &biter : nullptr)) {
-            done = true;
-            break;
+          s = biter.status();
+          if (s.ok() && done) {
+            referenced_data_size = biter.key().size() + biter.value().size();
```

Comment:
This could be updated when you set the flag to true.

Reply:
Thanks. I added the check on the block_cache_tracer_ here.

Reply:
I was thinking about moving the assignment of referenced_data size right after the assignment of does_referenced_key_exist. Those two should go hand in hand, right?

Reply:
Good catch!

Reply:
Sry I missed this comment earlier. I moved the referenced_data_size right after the assignment of does_referenced_key_exist.